### PR TITLE
[WIP] Win32 build tools

### PR DESCRIPTION
**Pull request by AchimTuran**
This PR will include wrapper scripts to simplify add-on development with Visual Studio. Furthermore it adds [jom](https://wiki.qt.io/Jom) to for faster binary add-on builds. It also adds [ninja](https://ninja-build.org/), which we can use to speed up our Windows build with jenkins.

During development I also noticed that multiple calls to `"%VS140COMNTOOLS%..\..\VC\bin\vcvars32.bat"` breaks an open command line window. To workaround I inserted an if statement.

### CODE REVIEW

**Comment by Montellese** [comment]
Please don't check in exe's into git. They should be downloaded before or during build like all other build tools.

**Comment by afedchin** [comment]
I think binaries in repo it's a bad idea.

**Comment by AchimTuran** [comment]
I fully agree with you @Montellese and @afedchin.

Could you mention me a place where this can be done properly?

File: tools/buildsteps/win32/bootstrap-addons.bat [hunk 1]
```diff
@@ -32,6 +35,7 @@ IF "%WORKDIR%" == "" (
 )
 
 rem setup some paths that we need later
+SET BUILD_ON_CORES=8
```
Comment (Paxxi) [inline tools/buildsteps/win32/bootstrap-addons.bat]: There is an environment variable for it in Windows but don't remember the name now

**Comment by Paxxi** [comment]
On my phone currently so won't comment about the code. 
Why include both the zip and exe? Also can we drop the changelogs?

File: tools/buildsteps/win32/bootstrap-addons.bat (hunk 1 above)
Comment (AchimTuran) [inline tools/buildsteps/win32/bootstrap-addons.bat]: Ah didn't know that. You mean %NUMBER_OF_PROCESSORS%?
Which is discussed [here](http://stackoverflow.com/questions/22919076/find-number-of-cpus-and-cores-per-cpu-using-command-prompt).

Comment (Paxxi) [inline tools/buildsteps/win32/bootstrap-addons.bat]: Yep that's the one

**Comment by AchimTuran** [comment]
As I'm not very familiar with our current build tools, I committed them all to show from where I get them.
Sure we can drop the changeling if it isn't needed. That's why I added WIP to the title.

**Comment by AchimTuran** [comment]
@afedchin @Montellese 
Would it be ok for you to download and extract jom and ninja during the run of [DownloadBuildDeps.bat](https://github.com/xbmc/xbmc/blob/master/project/BuildDependencies/DownloadBuildDeps.bat)?

**Comment by Montellese** [comment]
@AchimTuran yes that's how it should be done. And the downloaded files should be extracted to the subdirectories in `project/BuildDependencies/` like we already do for `wget.exe`, `doxygen.exe` et al.

**Comment by AchimTuran** [comment]
@Montellese 
> @AchimTuran yes that's how it should be done. And the downloaded files should be extracted to the  subdirectories in project/BuildDependencies/ like we already do for wget.exe, doxygen.exe et al.

Then we should add ninja and jom to http://mirrors.kodi.tv right? Do know who could do this?

**Comment by Montellese** [comment]
Unfortunately I don't know the new credentials for our FTP server. But I'm sure @MartijnKaijser does.

**Comment by Rechi** [comment]
@AchimTuran is this still something we want?

**Comment by AchimTuran** [comment]
I don't know if it is something what we will need. These scripts helped me a lot to properly build binary add-ons for windows. You can see them more like a workaround. Our build system for binary add-ons needs a refactoring.

**Comment by MartijnKaijser** [comment]
I will close this PR. If there's interest please open a new one (with required changes if needed).

**Pull request closed without merging**